stable_baselines3/sac/sac.py

```diff
@@ -202,6 +202,7 @@ def train(self, gradient_steps: int, batch_size: int = 64) -> None:
                 self.actor.reset_noise()
 
             # Action by the current actor for the sampled state
+            self.actor.set_training_mode(True)
```

Comment:
`self.actor` is just an alias of `self.policy.actor` so this (and the other calls to `critic`/`actor`) should not be needed, no?

Reply:
The reason for this is that on lines 229-232 we have 

```
with th.no_grad():
    # Select action according to policy
    self.actor.set_training_mode(False)
    next_actions, next_log_prob = self.actor.action_log_prob(replay_data.next_observations)
```

What's happening is that, when `self.actor` is being used to get the next actions without gradients, I'm putting it into eval mode, then on line 205 I put it back into train mode before calling

```
actions_pi, log_prob = self.actor.action_log_prob(replay_data.observations)
```

I _think_ that this is the correct behaviour. What do you think?

Reply:
As below, I've removed the extra calls to `set_training_mode`.